### PR TITLE
Added Datapackage class

### LINKED ISSUE #11: Implement DataPackage class
# Overview

```
async DataPackage(descriptor, profile='base', raise_invalid=True, remote_profiles=False)
    valid -> bool
    errors -> list
    descriptor -> dict
    update(**values)
    add_resource(descriptor)
    resources -> Resource[]
```

# Tasks

- [x] implement stub `DataPackage` class wich could be instantiated as `Promise`
- [x] add `constructor` method to store all arguments as internal state
  - [x] descriptor could be javascript object OR
  - [x] descriptor could be a link to json file (local or remote)
  - [x] call `validate(descriptor)` in constructor to validate datapackage descriptor
    - [x] if `raise_invalid` raise on invalid
    - [x] if not `raise_invalid` don't raise on invalid just store errors internally
- [x] implement `datapackage.valid` getter (always true if `raise_invalid` is true otherwise depends on validatation result)
- [x] implement `datapackage.errors` getter (always empty array if `raise_invalid` is true otherwise list of validation errors)
- [x] implement `descriptor` getter
- [x] implement `update` method 
  - [x] merge provided object to descriptor and re-validate (use `raise_invalid` logic here also)
  - [x] raise an error if user trying to alter `resources` key
- [x] implement `resources` getter returning array of earlier implemented `Resource` instances
- [x] implement `add_resource` method 
  - [x] should update descriptor
  - [x] should update resources
  - [x] should trigger datapackage revalidation (use `raise_invalid` logic here also)
- [ ] write comprehensive tests using `data` dir (real datapackage descriptors)

### PR DESCRIPTION
**Pull request by dumyan**
fixes #11 (waffle.io supported form) 

---

@roll please review

### CODE REVIEW

**Comment by roll** [comment]
@dumyan 
please rebase (or recreate PR) to remove profiles PR from it

**Comment by dumyan** [comment]
@roll 
Sorry for the commits baggage, I got it sorted now.

File: src/datapackage.js [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+import validate from './validate'
+import Resource from './resource'
+import Utils from './utils'
+
+/* Base class for working with datapackages */
+class DataPackage {
+  /**
+   * Create a Promise that will resolve with DataPackage instance.
+   *
+   * @param {Object} descriptor - a datapackage descriptor
```
Comment (roll) [inline src/datapackage.js]: object or url string?

File: src/datapackage.js [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+   * @param newDescriptor {Object}
+   * @return {Promise} Resolves in true or rejects with Array of errors
+   */
+  update(newDescriptor) {
+    // Check if altering resources but ignore if raiseInvalid is false
+    if (this._resourcesAreSame(newDescriptor) || !this._raiseInvalid) {
+      let mergedDescriptors
+
+      return new Promise((resolve, reject) => {
+        this._loadDescriptor(newDescriptor).then(theDescriptor => {
```
Comment (roll) [inline src/datapackage.js]: we don't need to support url for newDescriptor only {Object}

**Comment by dumyan** [comment]
@roll

Please review, I have made some other changes as well: simplified code, improved docstrings and testing

**Pull request merged**